edit_course/operations/configure.py

```diff
@@ -429,3 +468,54 @@ def configure_content(instance, url):
             category.delete()
 
     return errors
+
+def get_target_category(category, course=None, course_url=None):
+
+    course_name = None
+    instance_name = None
+
+
+    if not category:
+        return None, _("ExerciseCollection object requires collection_category.")
+
+    if (course is None) == (course_url is None):
+        return None, _("ExerciseCollection object must have either identified or URL")
+
+    if course:
+        course_name, instance_name = course.split(";")
+
+        try:
+            Course.objects.get(name=course_name)
+        except:
+            return None, _('Course: {} does not exist'.format(course_name))
+
+        try:
+            course_instance = CourseInstance.objects.get(instance_name=instance_name,
+                                                         course__name=course_name)
+        except ObjectDoesNotExist:
+            return None, _("Course: {}, Instance: {}, not found.".format(course_name, instance_name))
+
+    else:
+        course_slug_begin = 0
+        if not settings.BASE_URL in course_url:
```

Comment:
Why configuration url includes the site part? Kind of makes sense of course as they are site specifig.

Reply:
Copy pasted URLs usually contain also the site.
By checking that the URL is referring to the same system it's being run we can inform users that they can't use external services (like other A+ systems) for this.

I think giving specific error from wrong site in the URL is better than just informing that the course couldn't be found.

Reply:
Done side is that it doesn't support local testing. Though, local testing uses different url path and course name also, so it's problem any case :disappointed:

Reply:
We will go with this for now.

Reply:
Oh right... I forgot about the local testing of the course content.

Maybe the configuration could just skip adding the exercisecollections when doing local testing?
This could probably be done by checking if BASE_URL is '127.0.0.1' or 'localhost'?

Reply:
But then again that would make local testing generally impossible...
The only working solution right now is to do content testing without exercisecollections and test them separately on the production server.

Reply:
Settings are already different in repo, local containers and in production. See https://github.com/apluslms/run-aplus-front/blob/dev/Dockerfile#L5 for an example.